### PR TITLE
upgrade of kogito (0.3.0) that supports persistence based on infinispan

### PR DESCRIPTION
**Pull request by mswiderski**

### CODE REVIEW

**Comment by mswiderski** [comment]
Windows test failure does not seem to be related to his changes.

File: extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java [hunk 1]
```diff
@@ -150,10 +247,12 @@ private ApplicationGenerator createApplicationGenerator(Path projectPath, Launch
             throws IOException {
 
         Path srcPath = projectPath.resolve("src");
-        String appPackageName = "org.kie.kogito.app";
+        boolean usePersistence = combinedIndexBuildItem.getIndex()
+                .getClassByName(createDotName("org.kie.kogito.persistence.KogitoProcessInstancesFactory")) != null;
```
Comment (cescoffier) [inline extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java]: You may want to create a constant for this.

File: extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java [hunk 2]
```diff
@@ -69,16 +131,50 @@ public void generateModel(ArchiveRootBuildItem root,
...
+            },
+                    launchMode.getLaunchMode(), result, targetClassesPath);
+
+            Index index = kogitoIndexer.complete();
+
+            generatePersistenceInfo(root, generatedBeans, CompositeIndex.create(combinedIndexBuildItem.getIndex(), index),
+                    launchMode, resource);
+
+            reflectiveClass.produce(
+                    new ReflectiveClassBuildItem(true, true, "org.kie.kogito.services.event.ProcessInstanceDataEvent"));
```
Comment (cescoffier) [inline extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java]: Do you need method and field access?

**Review by cescoffier — CHANGES_REQUESTED** [review]
Mostly cosmetic.

We would need to update the quickstart also. Was the QS updated (and tested in native)?

File: extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java (hunk 2 above)
Comment (mswiderski) [reply extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java]: It is used by the jsonb Marshallers I I believe it is needed.

**Comment by mswiderski** [comment]
QuickStart was tested in all modes - there is an issue with dev mode but that is a general issue with json marshalling https://github.com/quarkusio/quarkus/issues/3624

In general QuickStart does not have to be updated

File: extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java (hunk 1 above)
Comment (mswiderski) [reply extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java]: updated

**Comment by mswiderski** [comment]
@cescoffier comments addressed, thanks for review

**Review by cescoffier — APPROVED** [review]

**Pull request merged**